statsmodels/tsa/statespace/mlemodel.py

```diff
@@ -159,6 +164,11 @@ def fit(self, start_params=None, transformed=True,
         return_params : boolean, optional
             Whether or not to return only the array of maximizing parameters.
             Default is False.
+        hessian_method : {'opg','oim','cs'}, optional
+            The method by which the Hessian is numerically approximated. 'opg'
+            uses outer product of gradients, 'oim' uses the information
+            matrix formula from Harvey (1989), and 'cs' uses second-order
```

Comment:
> about cov_type: LikelihoodModel is a base class and not all subclasses have the cov_type argument. I thought ARIMA doesn't have a covtype choice.
The final models for the user OLS, WLS, Logit, Poisson, ... and SARIMAX should have and explicit cov_type argument in fit. We can put it explicitly in the fit of a class if it is implemented for the subclasses. I don't know where in the class hierarchy this would be the case for the statespace models.

I will make this explicit in the `MLEModel`, since it should be appropriate for all state space models.

> hessian_method: AFAIU it's irrelevant or superseded by cov_type for inference
I guess it can be useful for the optimization, but if it's only for that, then I would make that more explicit and call it optim_hessian. (which I guess is only relevant if the optimizer uses the hessian. So maybe we should hide hessian_method in the kwargs with the other optimization options.

I changed the name to `optim_hessian` and added a note that it's only used if the optimizer uses the Hessian. I didn't subsume it in kwargs because the base class only checks if a `hessian` method is defined in the class, and `optim_hessian` is only used to tell the state space class which method to use. So it's not a generic optimization option. But I can hide it if you prefer.